niceman/distributions/piputils.py

```diff
@@ -38,6 +42,50 @@ def parse_pip_show(out):
     return pip_info
 
 
+def _pip_batched_show(session, which_pip, pkgs):
+    cmd = [which_pip, "show", "-f"]
+    batch = execute_command_batch(session, cmd, pkgs)
+    entries = (stacked.split("---") for stacked, _, _ in batch)
+
+    for pkg, entry in zip(pkgs, itertools.chain(*entries)):
+        info = parse_pip_show(entry)
+        yield pkg, info
```

Comment:
I would have appreciated a unittest for this function which mocks away `execute_command_batch` and demonstrate that it does work e.g. when output for multiple pkgs in a batch is output (on some dummified but still legit).  That could allow others comprehend on what is going on here (how pip show -f output should look like etc), if things start to break (e.g. `pip show -f` format changes or smth odd in it, like someone using `---` in the Summary or a url for some reason ;-))